File: Day5/main.js
function organizeShoes(shoes) {
  const shoeMap = new Map();

  for (const { type, size } of shoes) {
    if (!shoeMap.has(size)) {
      shoeMap.set(size, { I: 0, R: 0 });
    }
    shoeMap.get(size)[type]++;
  }

  const result = [];

  for (const [size, counts] of shoeMap) {
    const pairs = Math.min(counts.I, counts.R);
    for (let i = 0; i < pairs; i++) {
      result.push(size);
    }
  }

  return result.sort((a, b) => a - b);
}

File: Day5/main.py
def organize_shoes(shoes):
    shoe_map = {}

    for shoe in shoes:
        size = shoe['size']
        shoe_type = shoe['type']
        if size not in shoe_map:
            shoe_map[size] = {'I': 0, 'R': 0}
        shoe_map[size][shoe_type] += 1

    result = []

    for size, counts in shoe_map.items():
        pairs = min(counts['I'], counts['R'])
        result.extend([size] * pairs)

    return sorted(result)
